community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/planner/logical/idp/connectComponentsAndSolveOptionalMatch.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.compiler.v3_0.planner.logical.idp
+
+import org.neo4j.cypher.internal.compiler.v3_0.planner.QueryGraph
+import org.neo4j.cypher.internal.compiler.v3_0.planner.logical.{QueryPlannerKit, LogicalPlanningContext}
+import org.neo4j.cypher.internal.compiler.v3_0.planner.logical.plans.LogicalPlan
+
+import scala.annotation.tailrec
+
+case object connectComponentsAndSolveOptionalMatch {
```

Comment:
This object seems to be unused? It has the same name and signature as a `private def` above, but no calls are made in here?